src/intern.rs

```diff
@@ -18,7 +18,7 @@ use std::{
 /// using the same [`FeatureKey`] to get the string from two separate [`FeatureStorage`]s will likely result
 /// in two different strings (if [`FeatureStorage::get()`] doesn't return [`None`], that is).
 #[repr(transparent)]
-#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+#[derive(Clone, Copy, PartialEq, Eq, Debug, Hash)]
```

Comment:
This should not implement `Hash`, since it itself is a hash. (Hashing a hash is just extra work without extra benefit!)

Instead, `FeatureKey` should derive `Ord` and `PartialOrd` and be used in a `BTreeSet` instead of a `HashSet`. `BTreeSet` offers the same de-duplication capabilities as `HashSet`, but is faster because it doesn't hash the key.

Reply:
done